### ISSUE TITLE
Writing a reverse race condition

### ISSUE THREAD

**Issue by Vpr99** [redux-saga/redux-saga#1189]
I've been reading the documentation on [Racing Effects](https://github.com/redux-saga/redux-saga/blob/master/docs/advanced/RacingEffects.md) and I've got a question - what if I want to use the _slower_ of the 2 effects. How would you go about designing the saga flow for this?

Here's a scenario:

The user clicks a button which:

 * dispatches an api call
 * changes the UI state of the button to `loading`

Once the API returns:
 * dispatch the returned data
 * transition the UI state to `success`

Simple enough. However, for UX reasons, I want to show the `loading` state for a minimum of 750ms, so that the user has a chance to see that an action has occurred.

Basically, I want to hold off dispatching the `success` state until either the API resolves, or 750ms have passed, whichever is __slower__.

Does anybody have thoughts on how to approach the design of something like this?

**Comment by Vpr99** [comment]
Just realized I can probably tackle this with [`yield all`](https://github.com/redux-saga/redux-saga/tree/master/docs/api#alleffects---parallel-effects)

**Comment by Andarist** [comment]
@Vpr99 yeah, `all` should do the trick just fine :)
